aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/HostedUICognitoActions.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.actions
+
+import com.amplifyframework.auth.cognito.AuthEnvironment
+import com.amplifyframework.auth.cognito.helpers.JWTParser
+import com.amplifyframework.statemachine.Action
+import com.amplifyframework.statemachine.codegen.actions.HostedUIActions
+import com.amplifyframework.statemachine.codegen.data.SignInMethod
+import com.amplifyframework.statemachine.codegen.data.SignedInData
+import com.amplifyframework.statemachine.codegen.events.AuthenticationEvent
+import com.amplifyframework.statemachine.codegen.events.HostedUIEvent
+import java.util.Date
+
+object HostedUICognitoActions : HostedUIActions {
+
+    override fun showHostedUI(event: HostedUIEvent.EventType.ShowHostedUI) =
+        Action<AuthEnvironment>("InitHostedUIAuth") { id, dispatcher ->
+            logger?.verbose("$id Starting execution")
+            try {
+                if (hostedUIClient == null) throw Exception() // TODO: More detailed exception
```

Comment:
We have exceptions inside AuthException which can be thrown here. Worst case just throw a generic Auth Exception?

Reply:
Yeah some of these can be generic. I wanted to do an audit of ours/iOS together to match any error messaging that could be returned, so I threw todos anywhere I hadn't reviewed yet.

Reply:
Will be following up with a PR updating exception messaging.